src/Symfony/Component/Cache/Adapter/DoctrineDbalAdapter.php

```diff
@@ -0,0 +1,397 @@
+        if ('sqlsrv' === $platformName || 'oci' === $platformName) {
+            $stmt->bindParam(1, $id);
+            $stmt->bindParam(2, $id);
+            $stmt->bindParam(3, $data, ParameterType::LARGE_OBJECT);
+            $stmt->bindValue(4, $lifetime, ParameterType::INTEGER);
+            $stmt->bindValue(5, $now, ParameterType::INTEGER);
+            $stmt->bindParam(6, $data, ParameterType::LARGE_OBJECT);
+            $stmt->bindValue(7, $lifetime, ParameterType::INTEGER);
+            $stmt->bindValue(8, $now, ParameterType::INTEGER);
+        } elseif (null !== $platformName) {
+            $stmt->bindParam(1, $id);
+            $stmt->bindParam(2, $data, ParameterType::LARGE_OBJECT);
+            $stmt->bindValue(3, $lifetime, ParameterType::INTEGER);
+            $stmt->bindValue(4, $now, ParameterType::INTEGER);
+        } else {
+            $stmt->bindParam(1, $data, ParameterType::LARGE_OBJECT);
+            $stmt->bindValue(2, $lifetime, ParameterType::INTEGER);
+            $stmt->bindValue(3, $now, ParameterType::INTEGER);
+            $stmt->bindParam(4, $id);
+
+            $insertStmt = $this->conn->prepare($insertSql);
+            $insertStmt->bindParam(1, $id);
+            $insertStmt->bindParam(2, $data, ParameterType::LARGE_OBJECT);
+            $insertStmt->bindValue(3, $lifetime, ParameterType::INTEGER);
+            $insertStmt->bindValue(4, $now, ParameterType::INTEGER);
+        }
+
+        foreach ($values as $id => $data) {
+            try {
+                $result = $stmt->executeQuery();
+            } catch (TableNotFoundException $e) {
+                if (!$this->conn->isTransactionActive() || \in_array($platformName, ['pgsql', 'sqlite', 'sqlsrv'], true)) {
+                    $this->createTable();
+                }
+                $result = $stmt->executeQuery();
+            }
+            if (null === $platformName && 0 === $result->rowCount()) {
+                try {
+                    $insertStmt->executeStatement();
+                } catch (DBALException $e) {
+                    // A concurrent write won, let it be
+                }
+            }
+        }
+
+        return $failed;
+    }
+
+    private function getPlatformName(): string
+    {
+        if (isset($this->platformName)) {
+            return $this->platformName;
+        }
+
+        $platform = $this->conn->getDatabasePlatform();
+
+        switch (true) {
+            case $platform instanceof \Doctrine\DBAL\Platforms\MySQLPlatform:
+            case $platform instanceof \Doctrine\DBAL\Platforms\MySQL57Platform:
+                return $this->platformName = 'mysql';
+
+            case $platform instanceof \Doctrine\DBAL\Platforms\SqlitePlatform:
+                return $this->platformName = 'sqlite';
+
+            case $platform instanceof \Doctrine\DBAL\Platforms\PostgreSQL94Platform:
+            case $platform instanceof \Doctrine\DBAL\Driver\PDO\PgSQL\Driver:
```

Comment:
```suggestion
            case $platform instanceof \Doctrine\DBAL\Platforms\PostgreSQLPlatform:
```

`$platform` should never be a driver.

Reply:
Thanks for the review. I applied the patches and rebased the branch.